### ISSUE TITLE
No data found. Please verify arguments (train_data_dir must be the parent of folders with images

### ISSUE THREAD

**Issue by Shark1337** [bmaltais/kohya_ss#1725]
Greetings,

Whenever I try to train a model, I encounter this error, and I have a 4090.

```
23:17:46-526134 INFO     accelerate launch --num_cpu_threads_per_process=2 "./train_network.py" --enable_bucket
                         --min_bucket_reso=256 --max_bucket_reso=2048
                         --pretrained_model_name_or_path="CompVis/stable-diffusion-v1-4"
                         --train_data_dir="C:/images/118_DPITW" --resolution="512,512"
                         --output_dir="C:/Loras/RODENASDPIT/output" --network_alpha="1" --save_model_as=safetensors
                         --network_module=networks.lora --text_encoder_lr=5e-05 --unet_lr=0.0001 --network_dim=8
                         --output_name="DPITW" --lr_scheduler_num_cycles="10" --no_half_vae --learning_rate="0.0001"
                         --lr_scheduler="cosine" --train_batch_size="1" --save_every_n_epochs="1"
                         --mixed_precision="bf16" --save_precision="bf16" --cache_latents --optimizer_type="AdamW8bit"
                         --max_data_loader_n_workers="0" --bucket_reso_steps=64 --xformers --bucket_no_upscale
                         --noise_offset=0.0
prepare tokenizer
Using DreamBooth method.
prepare images.
0 train images with repeating.
0 reg images.
no regularization images / 正則化画像が見つかりませんでした
[Dataset 0]
  batch_size: 1
  resolution: (512, 512)
  enable_bucket: False


[Dataset 0]
loading image sizes.
0it [00:00, ?it/s]
make buckets
min_bucket_reso and max_bucket_reso are ignored if bucket_no_upscale is set, because bucket reso is defined by image size automatically / bucket_no_upscaleが指定された場合は、bucketの解像度は画像サイズから自動計算されるため、min_bucket_resoとmax_bucket_resoは無視されます
number of images (including repeats) / 各bucketの画像枚数（繰り返し回数を含む）
C:\kohya_ss\venv\lib\site-packages\numpy\core\fromnumeric.py:3464: RuntimeWarning: Mean of empty slice.
  return _methods._mean(a, axis=axis, dtype=dtype,
C:\kohya_ss\venv\lib\site-packages\numpy\core\_methods.py:192: RuntimeWarning: invalid value encountered in scalar divide
  ret = ret.dtype.type(ret / rcount)
mean ar error (without repeats): nan
No data found. Please verify arguments (train_data_dir must be the parent of folders with images) / 画像がありません。引数指定を確認してください（train_data_dirには画像があるフォルダではなく、画像があるフォルダの親フォルダを指定する必要があります）
```

**Comment by nokeyfan** [comment]
In lora (Dreamboothh never done); your images should be in the img folder. Like this: C:/images/img/118_DPITW

**Comment by belong-to** [comment]
I would like to know how to solve the problem of image regularization when training in kohyass

**Comment by belong-to** [comment]
2024-06-24 13:50:45 INFO     update token length: 75                                                  train_util.py:4244
                    INFO     Using DreamBooth method.                                               train_network.py:172
                    WARNING  ignore directory without repeats /                                       config_util.py:584
                             繰り返し回数のないディレクトリを無視します: img
                    WARNING  ignore directory without repeats /                                       config_util.py:584
                             繰り返し回数のないディレクトリを無視します: input
                    INFO     prepare images.                                                          train_util.py:1572
                    INFO     0 train images with repeating.                                           train_util.py:1613
                    INFO     0 reg images.                                                            train_util.py:1616
                    WARNING  no regularization images / 正則化画像が見つかりませんでした              train_util.py:1621
                    INFO     [Dataset 0]                                                              config_util.py:565
                               batch_size: 1
                               resolution: (512, 512)
                               enable_bucket: False
                               network_multiplier: 1.0


                    INFO     [Dataset 0]                                                              config_util.py:571
                    INFO     loading image sizes.                                                      train_util.py:853
0it [00:00, ?it/s]
                    INFO     make buckets                                                              train_util.py:859
                    WARNING  min_bucket_reso and max_bucket_reso are ignored if bucket_no_upscale is   train_util.py:876
                             set, because bucket reso is defined by image size automatically /
                             bucket_no_upscaleが指定された場合は、bucketの解像度は画像サイズから自動計
                             算されるため、min_bucket_resoとmax_bucket_resoは無視されます
                    INFO     number of images (including repeats) /                                    train_util.py:905
                             各bucketの画像枚数（繰り返し回数を含む）
D:\kohya_ss-master\venv\lib\site-packages\numpy\core\fromnumeric.py:3504: RuntimeWarning: Mean of empty slice.
  return _methods._mean(a, axis=axis, dtype=dtype,
D:\kohya_ss-master\venv\lib\site-packages\numpy\core\_methods.py:129: RuntimeWarning: invalid value encountered in scalar divide
  ret = ret.dtype.type(ret / rcount)
                    INFO     mean ar error (without repeats): nan                                      train_util.py:915
                    ERROR    No data found. Please verify arguments (train_data_dir must be the     train_network.py:212
                             parent of folders with images) /
                             画像がありません。引数指定を確認してください（train_data_dirには画像が
                             あるフォルダではなく、画像があるフォルダの親フォルダを指定する必要があ
                             ります）

**Comment by FollowTheChalk** [comment]
If you look into the line here, It'd be easier to get it:
"ERROR No data found. Please verify arguments (train_data_dir must be the train_network.py:212
parent of folders with images)"
Your "Image folder (containing training images subfolders)" should be the parent directory of the folder which holds dataset images.
For example, if your train data are stored in "C:/training/10_Katarina(League_of_legends) person/",
You gotta pick the parent directory ("C:/training/") as your Image folder, not "C:/training/10_Katarina(League_of_legends) person/".
